### PR TITLE
refactor `selectObjectContent` to ts

### PR DESCRIPTION
**Pull request by prakashsvmx**
refactor `selectObjectContent` to ts

### CODE REVIEW

File: src/internal/xml-parser.ts [hunk 1]
```diff
@@ -394,3 +397,144 @@ export function parseBucketVersioningConfig(xml: string) {
   const xmlObj = parseXml(xml)
   return xmlObj.VersioningConfiguration
 }
+
+export function parseSelectObjectContentResponse(res: unknown) {
```
Comment (trim21) [inline src/internal/xml-parser.ts:401]: I think we know res is buffer

File: src/internal/xml-parser.ts [hunk 2]
```diff
@@ -394,3 +397,144 @@ export function parseBucketVersioningConfig(xml: string) {
...
+            // Continuation message: Not sure if it is supported. did not find a reference or any message in response.
+            // It does not have a payload.
+            const warningMessage = `Un implemented event detected  ${messageType}.`
+            // eslint-disable-next-line no-console
+            console.warn(warningMessage)
+          }
+        } // eventType End
+      } // Event End
+    } // messageType End
+  } // Top Level Stream End
```
Comment (aldy505) [inline src/internal/xml-parser.ts:539]: Why do we even have to state end bracket comments? In every IDE or text editor, brackets are all collapsible. This is not a Dart code anyway.

File: src/internal/xml-parser.ts [hunk 3]
```diff
@@ -394,3 +397,144 @@ export function parseBucketVersioningConfig(xml: string) {
...
+
+    const totalMsgLength = totalByteLengthBuffer.readInt32BE()
+    const headerLength = headerBytesBuffer.readInt32BE()
+    const preludeCrcByteValue = preludeCrcBuffer.readInt32BE()
+
+    if (preludeCrcByteValue !== calculatedPreludeCrc) {
+      // Handle Header CRC mismatch Error
+      throw new Error(
+        `Header Checksum Mismatch, Prelude CRC of ${preludeCrcByteValue} does not equal expected CRC of ${calculatedPreludeCrc}`,
+      )
```
[the remark this answers is not in the document]
Comment (prakashsvmx) [reply src/internal/xml-parser.ts:442]: Yes, as discussed earlier, we can think adding it post migration.

File: docs/API.md [hunk 4]
```diff
@@ -1671,12 +1670,8 @@ const selectOpts = {
   requestProgress: { Enabled: true },
 }
 
-minioClient.selectObjectContent('bucketName', 'objectName', selectOpts, function (err, res) {
-  if (err) {
-    return console.log('Unable to process select object content.', err.message)
-  }
-  console.log('Success')
-})
+const res = minioClient.selectObjectContent('bucketName', 'objectName', selectOpts)
```
Comment (aldy505) [inline docs/API.md:1673]: I believe we're missing `await` keyword?

File: src/internal/client.ts [hunk 5]
```diff
@@ -2102,4 +2110,77 @@ export class TypedClient {
...
+      } else {
+        throw new TypeError('outputSerialization is required')
+      }
+    } else {
+      throw new TypeError('valid select configuration is required')
+    }
+
+    const method = 'POST'
+    let query = `select`
+    query += '&select-type=2'
```
Comment (aldy505) [inline src/internal/client.ts:2149]: I don't know why we need to append to string while we can just do

```ts
const query = 'select&select-type=2'
```

The string is not altered by anything afterwards, though.

File: docs/API.md (hunk 4 above)
Comment (prakashsvmx) [reply docs/API.md]: Yes. thank you :+1:

File: docs/API.md [hunk 6]
```diff
@@ -1643,18 +1643,17 @@ composePromise
 
 <a name="selectObjectContent"></a>
 
-### selectObjectContent(bucketName, objectName, selectOpts[, callback])
+### selectObjectContent(bucketName, objectName, selectOpt)
```
Comment (kaankabalak) [inline docs/API.md:1646]: Very minor issue here, shouldn't this be `selectOpts` as the table below refers to `selectOpts`?

```suggestion
### selectObjectContent(bucketName, objectName, selectOpts)
```

File: src/internal/xml-parser.ts [hunk 7]
```diff
@@ -394,3 +397,145 @@ export function parseBucketVersioningConfig(xml: string) {
   const xmlObj = parseXml(xml)
   return xmlObj.VersioningConfiguration
 }
+
+// Used only in selectObectContent API.
```
Comment (kaankabalak) [inline src/internal/xml-parser.ts:401]: Looks like there's a very minor typo here:

```suggestion
// Used only in selectObjectContent API.
```

File: docs/API.md (hunk 6 above)
Comment (prakashsvmx) [reply docs/API.md]: I have updated. thank you. :+1:

**Review by kaankabalak — APPROVED** [review]

**Pull request merged**